js-exercises/abbreviate-string/abbreviateString.js

```diff
@@ -0,0 +1,11 @@
+import { isString } from 'util';
+
+function abbreviateString(str) {
+  if (!isString(str) || !str.length) throw new Error(' Please use a valid string.');
+
+  const strArray = str.trim().split(' ');
```

Comment:
The problem that I thought of this approach is, we only need the first and last words. It doesn't make sense to split all the words into an array just to get first and last. All the operation seems redundant to me. 

Hence, my approach was to use `str.indexOf(' ')` to find the first word and `str.lastIndexOf(' ')` to find the last word. 

The `lastIndexOf` method searches for a match from the end of a string, so less operation in theory.